aspnetcore/fundamentals/servers/aspnet-core-module.md

```diff
@@ -57,23 +57,51 @@ The ASP.NET Core Module has to be installed in IIS on your servers and in IIS Ex
 
+# [ASP.NET Core 2.x](#tab/aspnetcore2x)
+
+The [Microsoft.AspNetCore.Server.IISIntegration](https://www.nuget.org/packages/Microsoft.AspNetCore.Server.IISIntegration/) package is included in the ASP.NET Core metapackages ([Microsoft.AspNetCore](https://www.nuget.org/packages/Microsoft.AspNetCore/) and [Microsoft.AspNetCore.All](xref:fundamentals/metapackage)). If you don't use one of the metapackages, install `Microsoft.AspNetCore.Server.IISIntegration` separately. This is an interoperability pack that reads environment variables broadcast by ANCM to set up your app. The environment variables provide configuration information such as the port to listen on. 
+
+---
+
 ### Call UseIISIntegration
 
+# [ASP.NET Core 1.x](#tab/aspnetcore1x)
+
 In your application's `Main` method, call the `UseIISIntegration` extension method on [`WebHostBuilder`](http://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNetCore/Hosting/WebHostBuilder/index.html#Microsoft.AspNetCore.Hosting.WebHostBuilder.md). 
 
 [!code-csharp[](aspnet-core-module/sample/Program.cs?name=snippet_Main&highlight=12)]
 
-The `UseIISIntegration` method looks for environment variables that ANCM sets, and it does nothing if they aren't found. This behavior facilitates scenarios like developing and testing on MacOS and deploying to a server that runs IIS.  While running on the Mac, Kestrel acts as the web server, but when the app is deployed to the IIS environment, it automatically hooks up to ANCM and IIS.
+# [ASP.NET Core 2.x](#tab/aspnetcore2x)
+
+The `UseIISIntegration` extension method on [`WebHostBuilder`](http://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNetCore/Hosting/WebHostBuilder/index.html#Microsoft.AspNetCore.Hosting.WebHostBuilder.md) is called automatically when you run with IIS.
 
-### Don't call UseUrls
+If you aren't using one of the ASP.NET Core metapackages, and the `Microsoft.AspNetCore.Server.IISIntegration` package is not installed, you get a runtime error. If you call `UseIISIntegration` explicitly, you get a compile time error if the package is not installed.
+
+---
 
-ANCM generates a dynamic port to assign to the back-end process. `IWebHostBuilder.UseIISIntegration` picks up this dynamic port and configures Kestrel to listen on `http://locahost:{dynamicPort}/`. This overwrites other URL configurations like calls to `IWebHostBuilder.UseUrls`. Therefore, you don't need to call `UseUrls` when you use ANCM. When you run the app without IIS, it listens on the default port number at `http://localhost:5000`.
+The `UseIISIntegration` method looks for environment variables that ANCM sets, and it does nothing if they aren't found. This behavior facilitates scenarios like developing and testing on macOS and deploying to a server that runs IIS.  While running on macOS, Kestrel acts as the web server, but when the app is deployed to the IIS environment, it automatically hooks up to ANCM and IIS.
```

Comment:
would it be worth writing macOS/Linux in both places?
I don't like the colloquial **hooks up**. Should you say connects? Can you make it two sentence?

While running on macOS/Linus, Kestrel acts as the web server.  When the app is deployed to the IIS environment, it automatically connects to ANCM and IIS.